### PR TITLE
Programmatic Scrypto SBOR docs

### PR DESCRIPTION
**Pull request by krzlabrdx**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 1]
```diff
@@ -3158,3 +3166,323 @@ components:
...
+    ProgrammaticScryptoSborValueMap:
+      allOf:
+        - $ref: "#/components/schemas/ProgrammaticScryptoSborValue"
+        - type: object
+          required:
+            - key_value_kind
+            - value_value_kind
+            - entries
+          properties:
+            key_value_kind:
```
Comment (dhedey) [inline src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:3426]: You're missing non-required/nullable `key_type_name` and `value_type_name` here

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 2]
```diff
@@ -3158,3 +3166,323 @@ components:
...
+              $ref: "#/components/schemas/ProgrammaticScryptoSborValueTypeKind"
+            value_value_kind:
```
Comment (dhedey) [inline src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:3428]: I believe these should be `key_kind` and `value_kind` not `key_value_kind` and `value_value_kind`

**Review by dhedey — CHANGES_REQUESTED** [review]
A few changes requested

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 3]
```diff
@@ -3158,3 +3166,323 @@ components:
...
+        - U8
+        - U16
+        - U32
+        - U64
+        - U128
+        - String
+        - Enum
+        - Array
+        - Tuple
+        - Map
```
[the remark this answers is not in the document]
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:3193]: `Bytes` added, comment added.

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 4]
```diff
@@ -3158,3 +3166,323 @@ components:
...
+          I64: "#/components/schemas/ProgrammaticScryptoSborValueI64"
+          I128: "#/components/schemas/ProgrammaticScryptoSborValueI128"
+          U8: "#/components/schemas/ProgrammaticScryptoSborValueU8"
+          U16: "#/components/schemas/ProgrammaticScryptoSborValueU16"
+          U32: "#/components/schemas/ProgrammaticScryptoSborValueU32"
+          U64: "#/components/schemas/ProgrammaticScryptoSborValueU64"
+          U128: "#/components/schemas/ProgrammaticScryptoSborValueU128"
+          String: "#/components/schemas/ProgrammaticScryptoSborValueString"
+          Enum: "#/components/schemas/ProgrammaticScryptoSborValueEnum"
+          Array: "#/components/schemas/ProgrammaticScryptoSborValueArray"
```
[the remark this answers is not in the document]
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:3263]: `Bytes` added.

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 5]
```diff
@@ -3158,3 +3166,323 @@ components:
               items:
                 $ref: "#/components/schemas/PublicKeyHash"
 
+    #
+    # Programmatic Scrypto SBOR Values
+    #
+
+    ProgrammaticScryptoSborValueTypeKind:
```
[the remark this answers is not in the document]
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml]: Renamed.

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 6]
```diff
@@ -3158,3 +3166,323 @@ components:
...
+              type: string
+    ProgrammaticScryptoSborValueEnum:
+      allOf:
+        - $ref: "#/components/schemas/ProgrammaticScryptoSborValue"
+        - type: object
+          required:
+            - discriminator
+            - fields
+          properties:
+            discriminator:
```
[the remark this answers is not in the document]
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml]: Fixed, good catch.

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml (hunk 1 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml]: Added.

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml (hunk 2 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml]: Fixed.

**Review by PawelPawelec-RDX — APPROVED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,8 @@ Release Date: _unreleased_
 - Fixed missing `RecordTopOfDbLedger` observer call in `LedgerTransactionsProcessor`.
 - Fixed invalid response model for HTTP 400 Bad Request responses on input parameter validation failure.
 - Return 400 with validation error instead of 500 if `from_ledger_state` `state_version` is beyond known ledger tip.
+- Added strongly-typed OAS definition for `programmatic_json`.
```
[the remark this answers is not in the document]
Comment (krzlabrdx) [reply CHANGELOG.md:14]: Moved.

**Review by dhedey — COMMENTED** [review]
Oops realised I never posted this review

**Pull request merged**